### ISSUE TITLE
no-irregular-whitespace will give an error to irregular-whitespace if skipTemplates is true but template literal line breaks are included

### ISSUE THREAD

**Issue by eai04191** [eslint/eslint#13852]
**Tell us about your environment**

* **ESLint Version: `7.13.0`**
* **Node Version: `v12.18.0`**
* **npm Version: `v6.14.4`**

**What parser are you using?: `default`**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "extends": ["eslint:recommended"],
    "env": {
        "es6": true,
        "browser": true,
        "jest": true,
        "node": true
    },
    "rules": {
        "no-unused-vars": "off",
        "no-irregular-whitespace": [
            "error",
            {
                "skipTemplates": true
            }
        ]
    }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const passed = `hoge　fuga`;
const error = `hoge　fuga
hoge　fuga`;

// \U3000 is included between all `hoge` and `fuga`.
// But ESLint only gives an error on the 3rd line.
```

```bash
yarn eslint .\src\no-irregular-whitespace.js
```

**What did you expect to happen?**

ESLint does not raise an error for irregular-whitespace in template literals.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  3:5  error  Irregular whitespace not allowed  no-irregular-whitespace
```

**Are you willing to submit a pull request to fix this bug?**

Probably difficult. I want to see if this is a bug or the correct behavior.

**Comment by mdjermanovic** [comment]
Hi @eai04191, thanks for the bug report!

I confirmed this bug in our [online demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2VcIjogW1xuICAgIFwiZXJyb3JcIixcbiAgICB7XG4gICAgICAgIFwic2tpcFRlbXBsYXRlc1wiOiB0cnVlXG4gICAgfVxuXSovXG5cbmNvbnN0IHBhc3NlZCA9IGBob2dl44CAZnVnYWA7XG5jb25zdCBlcnJvciA9IGBob2dl44CAZnVnYVxuaG9nZeOAgGZ1Z2FgO1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) and prepared a fix.
